### PR TITLE
Drop Support for ADOT Collector

### PR DESCRIPTION
**Pull request by khanhntd**
# Description of the issue
Drop support for ADOT Collector and delete equivalent submodules.

# Description of changes 
* Delete unnecessary folders creation for following OS: Debian, Linux, Windows
* Delete [AWS-OTEL-Collector from the THIRD-PARTY-LICENSE](https://github.com/aws/amazon-cloudwatch-agent/pull/522/files#diff-35fb5cbcccfd11b8078a48d04c510b2bc00ae80eeea9b682c0f2c9bb36462333)
* Delete unnecessary build stages for AWS Otel Collector before bundle up with CloudWatchAgent
* Remove flag and also corresponding function for OTEL Collector in the packaging
* [Delete submodules and keep the full clone (instead of shallow clone) in Integration Test when building CWAgent](https://github.com/aws/amazon-cloudwatch-agent/pull/522/files#diff-9b9f9a38b889c9dcb55ba0330baa69c8fdd62ac1100b6eb59b6fc44008f5d320R91)

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
For backward compatibility, I have run the start-up commands in the previous OS and CloudWatchAgent runs normal to me (including MacOS)

**Image for installation of OTEL-Collector bundle with CWAgent before and after the change**
**CloudWatchAgent Installation on Windows:**
* Before the change
image
image

* After the change
image
image

**CloudWatchAgent Installation on Linux:**
* Before the change
image

* After the change
image

**CloudWatchAgent Installation on Debian:**
* Before the change
image

* After the change
image
image

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make linter`

### CODE REVIEW

File: packaging/debian/preinst
Comment (khanhntd) [inline packaging/debian/preinst:25]: It should be safe to remove this since we had triggered the roll-back and also there has not been any customers who report this issue.

**Review by sethAmazon — APPROVED** [review]

**Comment by khanhntd** [comment]
Will on hold the merge until the final decision has been made .

**Review by sethAmazon — APPROVED** [review]

File: packaging/linux/amazon-cloudwatch-agent.spec
Comment (SaxyPandaBear) [inline packaging/linux/amazon-cloudwatch-agent.spec:103]: What we should do here is remove the aoc user if it exists since we don't need it anymore. We shouldn't just leave the user there on the host

**Review by SaxyPandaBear — CHANGES_REQUESTED** [review]
1. It's been over a month since the PR was first opened. Can you confirm that this still works as expected and provide evidence on Windows and Linux, minimum, MacOS if possible?

File: packaging/linux/amazon-cloudwatch-agent.spec
Comment (ymtaye) [reply packaging/linux/amazon-cloudwatch-agent.spec:103]: Thanks for the suggestion! I will make changes and also validate changes work as expected

**Comment by ymtaye** [comment]
> 1. It's been over a month since the PR was first opened. Can you confirm that this still works as expected and provide evidence on Windows and Linux, minimum, MacOS if possible?
image

image

**Review by SaxyPandaBear — APPROVED** [review]
LGTM

**Comment by SaxyPandaBear** [comment]
I think what we should do eventually as a follow up is remove the statuses for CWOC in the control script, but probably don't need to do that right now.

**Comment by khanhntd** [comment]
> I think what we should do eventually as a follow up is remove the statuses for CWOC in the control script, but probably don't need to do that right now.


Hmm I removed that and remove some sanity check relates to that status but I guess we have decided in other way in QS.

**Pull request merged**
